src/admin-views/help-hub/resources-sidebar.php

```diff
@@ -0,0 +1,16 @@
+<?php
+/**
+ * The template that displays the resources sidebar.
+ *
+ * @var Tribe__Template $this  The template object.
+ * @var bool $is_opted_in      Whether the user has opted in to telemetry.
+ * @var bool $is_license_valid Whether the user has any valid licenses.
+ */
+
+if ( ! $is_license_valid ) {
+	$this->template( 'help-hub/resources-sidebar-no-license' );
+} elseif ( $is_opted_in ) {
+	$this->template( 'help-hub/resources-sidebar-has-license-has-consent' );
+} else {
+	$this->template( 'help-hub/resources-sidebar-has-license-no-consent' );
+}
```

Comment:
Most of the time, we just add all the templates, then add the logic at the beginning of the templates to bail early.

For example, at the top of `help-hub/resources-sidebar-no-license`:
```
if ( $is_license_valid ) {
	return;
}
```
...at the top of `help-hub/resources-sidebar-has-license-has-consent`:
```
if ( $is_license_valid && ! $is_opted_in ) {
	return;
}
```
...and so on and so forth...
```suggestion
$this->template( 'help-hub/resources-sidebar-no-license' );
$this->template( 'help-hub/resources-sidebar-has-license-has-consent' );
$this->template( 'help-hub/resources-sidebar-has-license-no-consent' );
```

Reply:
Yeah I've seen that in general, we do it in a lot of functions/filters too. I tend to operate the other way for a few reasons.

1. Less memory less overhead to put the logic above, instead of loading 1,2,3 other stacks to get to the correct location.
2. You see the branching logic in one place to understand things at a single glance. It's easier to reason and troubleshoot.
3. If I want to load that template somewhere else (or run that function, speaking generally) I can't unless I have the same context applied. It makes our "unit" tests more difficult to mock or our code more difficult to reuse, because we bury context. Similar to MVC the M should do a "thing" and the C should do the "branching / request context logic".

 @bordoni @codingmusician  - thoughts on above?